include/boost/json/detail/impl/number.ipp

```diff
@@ -636,16 +694,32 @@ finish(
         break;
 
     case state::exp3:
-        exp_ += off_;
-        if(eneg_)
+    {
+        if (eneg_)
             exp_ = -exp_;
+
+        if (pos_ == 0)
+        {
+            // this indicates that the decimal point preceeds any significant digits
+            // which means the mantissa has a magnitude < 1.0
```

Comment:
Consider just `// mantissa < 1`

Reply:
will change to `// |mantissa| < 1.0` as this succinctly contains 2 more data:

1. we are considering the magnitude of the mantissa
2. the magnitude of the mantissa is a real number